app/concerns/agent_controller_concern.rb

```diff
@@ -39,45 +39,47 @@ def validate_control_action
 
   def control!
     control_targets.each { |target|
-      begin
-        case control_action
-        when 'run'
-          case
-          when target.cannot_be_scheduled?
-            error "'#{target.name}' cannot run without an incoming event"
-          when target.disabled?
-            log "Agent run ignored for disabled Agent '#{target.name}'"
-          else
-            Agent.async_check(target.id)
-            log "Agent run queued for '#{target.name}'"
-          end
-        when 'enable'
-          case
-          when target.disabled?
-            target.update!(disabled: false)
-            log "Agent '#{target.name}' is enabled"
-          else
-            log "Agent '#{target.name}' is already enabled"
-          end
-        when 'disable'
-          case
-          when target.disabled?
-            log "Agent '#{target.name}' is alread disabled"
+      interpolate_with('target' => target) {
```

Comment:
Did you intentionally use brackets over `do/end` here? I generally follow `bbatsov`s [style guide](https://github.com/bbatsov/ruby-style-guide) to use `do/end` for multi line blocks, but don't feel strongly since there already is a block using `{}` here.